completions/asdf.fish

```diff
@@ -47,7 +47,7 @@ function __fish_asdf_plugin_list_all
 end
 
 function __fish_asdf_list_shims
-    ls $asdf_data_dir/shims
+    basename $asdf_data_dir/shims/*
```

Comment:
Nice one, I hadn't noticed the `path` builtin yet. Using `path basename` solves the nullglob issue as well. 

One thing to note is that the `path` builtin is relatively new, introduced in version 3.5 last year. That said, fish is usually always installed by users rather than being part of the OS, which means it's more likely to be up to date. Add that to the fact that completions are non-critical for `asdf` to function, I think it's safe for now.

Reply:
That's a good point - we're still figuring out which versions of which shells to support in #1436. 3.5 [is available](https://pkgs.org/download/fish) on Fedora 38 and 37, but not Debian 11 Bullseye or Ubuntu 22.04. @jthegedus might have a better idea about the versions of Fish we have historically supported

Reply:
While using `path basename` in this situation is convenient, theres no reason a loop (like for the bash implementation) wouldn't work if the decision is that older fish versions need to be supported. @jthegedus let me know what your decision is, and if needs be I'll happily change the implementation.